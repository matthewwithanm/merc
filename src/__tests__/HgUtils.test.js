/*
 * Copyright (c) 2016-present, Facebook, Inc.
 * All rights reserved.
 * @flow
 * @format
 */

/* eslint-env jest */

import fsPromise from 'nuclide-commons/fsPromise';
import path from 'path';
import {
  getRepoRoot,
  getCurrentRevisionHash,
  getMergeBaseHash,
} from '../HgUtils';

describe('getRepoRoot', () => {
  test('it reports the root', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo1');
    const dir = path.join(repoRoot, 'a/b');
    const actual = await getRepoRoot(dir).toPromise();
    expect(actual).toBe(repoRoot);
  });

  test("it returns null when there's no repo", async () => {
    const dir = path.resolve(__dirname, './fixtures/not-a-repo/a/b');
    const actual = await getRepoRoot(dir).toPromise();
    expect(actual).toBe(null);
  });
});

describe('getCurrentRevisionHash', () => {
  test('it returns the current revision hash', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo2');
    const hash = await getCurrentRevisionHash(repoRoot).toPromise();
    expect(hash).toMatchSnapshot();
  });
});

describe('getMergeBaseHash', () => {
  test('it returns the current revision hash', async () => {
    const repoRoot = path.resolve(__dirname, './fixtures/repo2');
    const hash = await getMergeBaseHash(repoRoot).toPromise();
    expect(hash).toMatchSnapshot();
  });
});
