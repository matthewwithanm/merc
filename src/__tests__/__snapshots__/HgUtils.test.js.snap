// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCurrentRevisionHash it returns the current revision hash 1`] = `"f670616d11764c7b0869c1fe769423e8cd71c537"`;

exports[`getMergeBaseHash it returns the current revision hash 1`] = `"c42dae43c973a141cf0f15d9bcb6f32093e1c1d2"`;
